internal/v2/api/client/debug_transport.go

```diff
@@ -0,0 +1,86 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// DebugTransport wraps an http.RoundTripper and logs response bodies on error.
+type DebugTransport struct {
+	Transport http.RoundTripper
+}
+
+// RoundTrip implements the http.RoundTripper interface.
+func (t *DebugTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Check if we're running acceptance tests
+	isAccTest := os.Getenv("TF_ACC") != ""
+
+	// Perform the request
+	resp, err := t.Transport.RoundTrip(req)
+	if err != nil {
+		// Log transport-level errors
+		logMessage := fmt.Sprintf("HTTP Transport Error: %v", err)
+		tflog.Error(req.Context(), logMessage)
+
+		// When running acceptance tests, output to stderr for test visibility
+		if isAccTest {
+			fmt.Fprintf(os.Stderr, "\n[DEBUG] %s\n", logMessage)
+		}
+
+		return resp, err
+	}
+
+	// If the status code indicates an error, log the response body
+	if resp.StatusCode >= 400 {
+		// Read the response body
+		bodyBytes, readErr := io.ReadAll(resp.Body)
+		if readErr != nil {
+			// If we can't read the body, log that issue and return the original response
+			logMessage := fmt.Sprintf("Failed to read error response body: %v", readErr)
+			tflog.Error(req.Context(), logMessage)
+
+			if isAccTest {
+				fmt.Fprintf(os.Stderr, "\n[DEBUG] %s\n", logMessage)
```

Comment:
for acceptance tests, print to stdout.
tflog isn't visible when running the acceptance tests.